File: frontend/types/subscription.ts
export type SubscriptionStatus = 'active' | 'expired' | 'cancelled' | 'paused';

export interface SubscriptionPlan {
  id: string;
  name: string;
  price: number;
  duration_days: number;
  description: string;
  features: string[];
}

export interface Subscription {
  id: string;
  user_id: string;
  plan_id: string;
  status: SubscriptionStatus;
  start_date: string;
  end_date: string;
  created_at: string;
  plan?: SubscriptionPlan;
}

File: backend/app/schemas/subscription.py
from pydantic import BaseModel
from typing import Optional, List

class SubscriptionPlanResponse(BaseModel):
    id: str
    name: str
    price: float
    duration_days: int
    description: str
    features: List[str]

class CreateSubscriptionRequest(BaseModel):
    plan_id: str

class SubscriptionResponse(BaseModel):
    id: str
    user_id: str
    plan_id: str
    status: str # active | expired | cancelled | paused
    start_date: str
    end_date: str
    created_at: Optional[str] = None
    plan: Optional[SubscriptionPlanResponse] = None
